src/handle/compile.ml

```diff
@@ -21,11 +21,15 @@ let parse_file : string -> Syntax.ast = fun fname ->
   | true  -> Parser.parse_file fname
   | false -> Parser.Dk.parse_file fname
 
-(** [compile force mp] compiles the file corresponding to [mp], when it is
-    necessary (the corresponding object file does not exist,  must be updated,
-    or [force] is [true]).  In that case,  the produced signature is stored in
-    the corresponding object file. *)
-let rec compile : bool -> Path.t -> Sign.t = fun force mp ->
+(** [with_handle ~handle ~force mp] compiles the file corresponding to module
+    path [mp] using function [~handle] to process commands. Module [mp] is
+    process when it is necessary, i.e. the corresponding object file does not
+    exist, or it must be updated, or [~force] is [true]). In that case, the
+    produced signature is stored in the corresponding object file. *)
+let rec with_handle :
```

Comment:
Why do you want to rename it into "with_handle"? Note that the output type is "compiler".

Reply:
I instead suggest the following renamings:
- handle -> proof_data_of_command
- with_proofs -> handle
- with_handle -> compile

Reply:
I renamed `handle` into `get_proof_data` (which is a bit shorter, and as explicit provided we use `Command.get_proof_data`).

Reply:
What about with_handle -> compile_with ?